store/rootmulti/store.go

```diff
@@ -438,8 +439,19 @@ func (rs *Store) LatestVersion() int64 {
 // LastCommitID implements Committer/CommitStore.
 func (rs *Store) LastCommitID() types.CommitID {
 	if rs.lastCommitInfo == nil {
+		emptyHash := sha256.Sum256([]byte{})
+		appHash := emptyHash[:]
 		return types.CommitID{
 			Version: GetLatestVersion(rs.db),
+			Hash:    appHash, // set empty apphash to sha256([]byte{}) if info is nil
+		}
+	}
+	if len(rs.lastCommitInfo.CommitID().Hash) == 0 {
```

Comment:
Refactoring `CommitInfo.Hash()` more makes sense to me. We can avoid extra checkings

Reply:
```suggestion
func (ci CommitInfo) Hash() []byte {
	// we need a special case for empty set, as SimpleProofsFromMap requires at least one entry
	if len(ci.StoreInfos) == 0 {
		return nil
	}

	rootHash, _, _ := maps.ProofsFromMap(ci.toMap())
	return rootHash
}
```

func (ci CommitInfo) Hash() []byte {
	// we need a special case for empty set, as SimpleProofsFromMap requires at least one entry
	if len(ci.StoreInfos) == 0 {
		return sha256.Sum256([]byte{})
	}

	rootHash, _, _ := maps.ProofsFromMap(ci.toMap())
	return rootHash
}

Reply:
there is no way to make it private, so adding Hash() without making private feels mute. Ill make the change but worried someone will still abuse it

Reply:
this would be quite api breaking, id say for now lets leave it since store v2 is around the corner.